packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -320,25 +298,7 @@ export class TokenRatesController extends StaticIntervalPollingController<TokenR
   #subscribeToNetworkStateChange() {
     this.messenger.subscribe(
       'NetworkController:stateChange',
-      // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-      // eslint-disable-next-line @typescript-eslint/no-misused-promises
-      async ({ networkConfigurationsByChainId }, patches) => {
-        const chainIdAndNativeCurrency: {
-          chainId: Hex;
-          nativeCurrency: string;
-        }[] = Object.values(networkConfigurationsByChainId).map(
-          ({ chainId, nativeCurrency }) => {
-            return {
-              chainId: chainId as Hex,
-              nativeCurrency,
-            };
-          },
-        );
-
-        if (this.#pollState === PollState.Active) {
-          await this.updateExchangeRates(chainIdAndNativeCurrency);
-        }
```

Comment:
I removed all this code because this was never being called, as `#pollState` was always disabled.

Assets for specific chains are updated individually in the UI (at least in extension) when the network filter is changed or a new network is added.

So the only thing this handler has been doing was remove chain data when deleting chains.